### DISCUSSION TITLE
Can virtualgl be used to run a whole desktop?

### DISCUSSION THREAD

**Discussion by corbinwunderlich** [Xpra-org/xpra#4410] [Q&A] [answered]
This would allow compositing and all that stuff.

**Comment by totaam** [comment, marked as answer]
Yes:
```
xpra desktop --exec-wrapper=vglrun fluxbox
```

This is [documented](https://github.com/Xpra-org/xpra/blob/master/docs/Usage/OpenGL.md).

> This would allow compositing and all that stuff.

All virtual screens used by xpra support compositing (except for `--pixel-depth=8`).
OpenGL and compositing are not directly related, though many compositing window managers do use OpenGL for performance.
